src/Modules/UngradedModules.jl

```diff
@@ -4985,13 +4993,30 @@ end
 
 Return the composition `b` $\circ$ `a`.
 """
-function *(h::ModuleFPHom, g::ModuleFPHom)
+function *(h::ModuleFPHom{T1, T2, Nothing}, g::ModuleFPHom{T2, T3, Nothing}) where {T1, T2, T3}
   @assert codomain(h) === domain(g)
   return hom(domain(h), codomain(g), Vector{elem_type(codomain(g))}([g(h(x)) for x = gens(domain(h))]))
 end
 
+function *(h::ModuleFPHom{T1, T2, <:Hecke.Map}, g::ModuleFPHom{T2, T3, <:Hecke.Map}) where {T1, T2, T3}
+  @assert codomain(h) === domain(g)
+  return hom(domain(h), codomain(g), Vector{elem_type(codomain(g))}([g(h(x)) for x = gens(domain(h))]), compose(base_ring_map(h), base_ring_map(g)))
+end
+
+function *(h::ModuleFPHom{T1, T2, <:Any}, g::ModuleFPHom{T2, T3, <:Any}) where {T1, T2, T3}
+  @assert codomain(h) === domain(g)
+  return hom(domain(h), codomain(g), 
+             Vector{elem_type(codomain(g))}([g(h(x)) for x = gens(domain(h))]), 
+             Hecke.MapFromFunc(x->(base_ring_map(g)(base_ring_map(h)(x))), 
+                                   base_ring(domain(h)), 
+                                   base_ring(codomain(g)))
+            )
+
+end
+
 compose(h::ModuleFPHom, g::ModuleFPHom) = h*g
 
+# TODO: We need to also make all of the code below sensitive to base changes!
```

Comment:
@jankoboehm : I think this is another issue by itself. But should be addressed!

Reply:
Think we can postpone until needed.